java/connectors/semantickernel-connectors-memory-redis/src/main/java/com/microsoft/semantickernel/connectors/memory/redis/RedisMemoryStore.java

```diff
@@ -0,0 +1,570 @@
+package com.microsoft.semantickernel.connectors.memory.redis;
+
+import com.microsoft.semantickernel.ai.embeddings.Embedding;
+import com.microsoft.semantickernel.memory.MemoryException;
+import com.microsoft.semantickernel.memory.MemoryException.ErrorCodes;
+import com.microsoft.semantickernel.memory.MemoryRecord;
+import com.microsoft.semantickernel.memory.MemoryStore;
+import reactor.core.publisher.Mono;
+import reactor.util.function.Tuple2;
+import reactor.util.function.Tuples;
+import redis.clients.jedis.JedisPooled;
+import redis.clients.jedis.exceptions.JedisDataException;
+import redis.clients.jedis.resps.ScanResult;
+import redis.clients.jedis.search.*;
+import redis.clients.jedis.search.schemafields.VectorField.VectorAlgorithm;
+import javax.annotation.Nonnull;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.security.InvalidParameterException;
+import java.text.MessageFormat;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.*;
+
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Semantic Memory implementation using Redis Vector Search. For more information about Redis
+ * Vector Search {@see https://redis.com/solutions/use-cases/vector-database/}
+ */
+public class RedisMemoryStore implements MemoryStore {
+
+    private static final VectorAlgorithm DefaultIndexAlgorithm = VectorAlgorithm.HNSW; // FLAT or HNSW
+    private static final String DefaultVectorType = RedisVectorType.FLOAT32; // prefer better accuracy
+    private static final String DefaultDistanceMetric = RedisVectorDistanceMetric.L2;
+    private static final int DefaultQueryDialect = 2; // this has to be >= 2 for vector functionality
+    private static final Integer DefaultVectorSize = 1536; // framework model dependent - ada-002
+    private static final String SUFFIX = "sk";
+    private static final String INDEX = "idx";
+    private static final String OK = "OK";
+
+    private final JedisPooled client;
+    private int vectorSize = DefaultVectorSize;
+    private VectorAlgorithm vectorIndexAlgorithm = DefaultIndexAlgorithm;
+    private String vectorDistanceMetric = DefaultDistanceMetric;
+    private String vectorType = DefaultVectorType;
+    private int queryDialect;
+
+    /**
+     * Create a new instance of semantic memory using Redis.
+     *
+     * @param client A Redis Database client connection.
+     * @param vectorSize Embedding vector size, defaults to 1536.
+     * @param vectorIndexAlgorithm Indexing algorithm for vectors, defaults to "HNSW"
+     * @param vectorDistanceMetric Metric for measuring vector distances, defaults to "COSINE"
+     * @param queryDialect Query dialect, must be 2 or greater for vector similarity searching, defaults to 2
+     */
+    public RedisMemoryStore(
+        JedisPooled client,
+        Integer vectorSize,
```

Comment:
Integer here and int on the connectionString variant of the ctor. Did you intend to use Integer here? I see in your comments that the vector sized defaults to 1536, but I don't see that being applied here. Nor for any of the other defaults. 

Also, would it make sense to put a maximum bound on vector size and query dialect?

Reply:
All are now Integers, and the constructors check range on those values